src/main/java/com/contrastsecurity/sdk/ContrastSDK.java

```diff
@@ -552,7 +577,13 @@ public InputStream makeRequest(HttpMethod method, String path) throws IOExceptio
     }
 
     public HttpURLConnection makeConnection(String url, String method) throws IOException {
-        HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+        HttpURLConnection connection;
+        if (this.proxy == null) {

```

Comment:
In the constructor for ContrastSDK without Proxy (https://github.com/Contrast-Security-OSS/contrast-sdk-java/pull/16/files?diff=split#diff-f40b6dbe37b70610f50eaf318983c50dR136), set `self.proxy` to Proxy.NO_PROXY.

Then, you can do-away with this `if` check. The `if` becomes

`connection = (HttpURLConnection) new URL(url).openConnection(this.proxy);`

And since this.proxy is set to NO_PROXY, it'll just pass through. See: https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#openConnection(java.net.Proxy)

Reply:
Done